test/controllers/TransferControllerTest.scala

```diff
@@ -390,12 +389,38 @@ class TransferControllerTest extends ControllerBaseSpec {
 
   "finished" should {
     "return success" when {
-      "A notification record is returned" in {
+      "make sure method remove never was called before" in {
+        verify(mockCachingService, times(0)).remove()(any(), any())
+      }
+
+      "A notification record is returned and cache is called" in {
         when(mockTransferService.getFinishedData(any())(any(), any()))
           .thenReturn(RelationshipRecordData.notificationRecord)
+
         val result = controller().finished()(request)
         status(result) shouldBe OK
       }
+
+      "make sure method remove was called only once" in {
+        verify(mockCachingService, times(1)).remove()(any(), any())
```

Comment:
Unit tests need to be independent. This test specifically requires previous test orders.